apis/v1alpha1/types.go

```diff
@@ -278,39 +296,68 @@ type HyperParameterSpecification struct {
-	FailureReason        *string            `json:"failureReason,omitempty"`
-	TrainingEndTime      *metav1.Time       `json:"trainingEndTime,omitempty"`
-	TrainingJobARN       *string            `json:"trainingJobARN,omitempty"`
-	TrainingJobName      *string            `json:"trainingJobName,omitempty"`
-	TrainingJobStatus    *string            `json:"trainingJobStatus,omitempty"`
-	TrainingStartTime    *metav1.Time       `json:"trainingStartTime,omitempty"`
-	TunedHyperParameters map[string]*string `json:"tunedHyperParameters,omitempty"`
+	CreationTime                                *metav1.Time                                 `json:"creationTime,omitempty"`
+	FailureReason                               *string                                      `json:"failureReason,omitempty"`
+	FinalHyperParameterTuningJobObjectiveMetric *FinalHyperParameterTuningJobObjectiveMetric `json:"finalHyperParameterTuningJobObjectiveMetric,omitempty"`
+	ObjectiveStatus                             *string                                      `json:"objectiveStatus,omitempty"`
+	TrainingEndTime                             *metav1.Time                                 `json:"trainingEndTime,omitempty"`
+	TrainingJobARN                              *string                                      `json:"trainingJobARN,omitempty"`
+	TrainingJobDefinitionName                   *string                                      `json:"trainingJobDefinitionName,omitempty"`
+	TrainingJobName                             *string                                      `json:"trainingJobName,omitempty"`
+	TrainingJobStatus                           *string                                      `json:"trainingJobStatus,omitempty"`
+	TrainingStartTime                           *metav1.Time                                 `json:"trainingStartTime,omitempty"`
+	TunedHyperParameters                        map[string]*string                           `json:"tunedHyperParameters,omitempty"`
+	TuningJobName                               *string                                      `json:"tuningJobName,omitempty"`
+}
+
+type HyperParameterTuningJobConfig struct {
+	HyperParameterTuningJobObjective *HyperParameterTuningJobObjective `json:"hyperParameterTuningJobObjective,omitempty"`
+	ParameterRanges                  *ParameterRanges                  `json:"parameterRanges,omitempty"`
+	ResourceLimits                   *ResourceLimits                   `json:"resourceLimits,omitempty"`
+	Strategy                         *string                           `json:"strategy,omitempty"`
+	TrainingJobEarlyStoppingType     *string                           `json:"trainingJobEarlyStoppingType,omitempty"`
+	TuningJobCompletionCriteria      *TuningJobCompletionCriteria      `json:"tuningJobCompletionCriteria,omitempty"`
 }
 
 type HyperParameterTuningJobObjective struct {
 	MetricName *string `json:"metricName,omitempty"`
+	Type       *string `json:"type_,omitempty"`
```

Comment:
This is interesting... the name of the field in the JSON output is `type_` for some reason...